packages/calendar-utils/src/calendar-body/calendar-body.styles.ts

```diff
@@ -64,31 +67,37 @@ const getIconFontColor = (props: TCalendarBody) => {
 
 const getCalendarIconContainerStyles = (
   props: TCalendarBody,
-  state: TState
+  state: TState,
+  isNewTheme: boolean
 ) => {
-  return css`
-    align-items: center;
-    box-sizing: border-box;
-    background: none;
-    border: 0;
-    border-left: 1px solid ${designTokens.borderColorForInput};
-    border-top-right-radius: ${designTokens.borderRadiusForInput};
-    border-bottom-right-radius: ${designTokens.borderRadiusForInput};
-    border-color: ${getIconBorderColor(props, state)};
-    color: ${getIconFontColor(props)};
-    cursor: ${props.isDisabled ? 'not-allowed' : 'default'};
-    height: 100%;
-    display: flex;
-    padding: ${designTokens.spacing10};
-    outline: 0;
-    transition: color ${designTokens.transitionStandard},
-      border-color ${designTokens.transitionStandard};
-    &:active,
-    &:hover:not(:disabled)&:not(:read-only),
-    &:focus {
-      border-color: ${designTokens.borderColorForInputWhenFocused};
-    }
-  `;
+  return [
+    css`
+      align-items: center;
+      box-sizing: border-box;
+      background: none;
+      border: 0;
+      border-top-right-radius: ${designTokens.borderRadiusForInput};
+      border-bottom-right-radius: ${designTokens.borderRadiusForInput};
+      border-color: ${getIconBorderColor(props, state)};
+      color: ${getIconFontColor(props)};
+      cursor: ${props.isDisabled ? 'not-allowed' : 'default'};
+      height: 100%;
+      display: flex;
+      padding: ${designTokens.spacing10};
+      outline: 0;
+      transition: color ${designTokens.transitionStandard},
+        border-color ${designTokens.transitionStandard};
+      &:active,
+      &:hover:not(:disabled)&:not(:read-only),
+      &:focus {
+        border-color: ${designTokens.borderColorForInputWhenFocused};
+      }
+    `,
+    !isNewTheme &&
+      css`
+        border-left: 1px solid ${getIconBorderColor(props, state)};
+      `,
+  ];
```

Comment:
Here I just extracted the left border of the icon so it's only rendered in the default theme.